src/core/utils/CorePlugin.ts

```diff
@@ -0,0 +1,30 @@
+import { JWPlugin } from '../JWPlugin';
+import { VDocument } from '../stores/VDocument';
+import { VNode } from '../stores/VNode';
+import { Action } from '../types/Flux';
+
+export class CorePlugin extends JWPlugin {
+    vDocument: VDocument;
```

Comment:
We had a discussion yesterday as to where to put the document, and I think no plugin should have direct access to the document, which is our actual state. We need to discuss about when we start creating the commitState step.